backend/Rhyolite/Backend/Email.hs

```diff
@@ -107,6 +117,44 @@ sendEmailFrom name' email recipients sub body =
 
 deriveNewtypePersistBackend (\m -> [t| EmailT $m |]) (\m -> [t| ReaderT EmailEnv $m |]) 'EmailT 'unEmailT
 
+-- | Build and send an email using a StaticWidget that can use frontend routes.
+sendWidgetEmailFrom
+  :: forall k (x :: k) a t r br m.
+     (MonadIO m, MonadEmail m, Reflex t)
+  => Text
+  -- ^ Name to use in the "from:" field.
+  -> Text
+  -- ^ Email address to use in the "from:" field.
+  -> NonEmpty Text
+  -- ^ List of recipients
+  -> Text
+  -- ^ Subject line
+  -> Text
+  -- ^ Base URL to build URLs for
+  -> (Encoder Identity Identity (R (Sum br (ObeliskRoute r))) PageName)
+  -- ^ Encoder to use for routes; usually the project's checkedRouteEncoder
+  -> Maybe ((RouteToUrlT (R r) Identity) Text)
+  -- ^ Body plaintext, with route decoder
+  -> SetRouteT t (R r) (RouteToUrlT (R r) (StaticWidget x)) a
+  -- ^ Body widget for the email
+  -> m ()
```

Comment:
I feel like at this point a record is worth it...

Reply:
https://hackage.haskell.org/package/named anyone?

Reply:
I like it in principle, but it's still both 1) a dependency and 2) a new "idiom." These make it a harder sell for a general purpose library.

Reply:
I had considered bundling the sender and routes information into something we could build off of a config to get that noise down, but I can see just wrapping the whole thing in a record too. I don't have a strong opinion on that.